src/functions-and-arrays.js

```diff
@@ -1,119 +1,272 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(n1, n2) {
+  if (n1 > n2) {
+    return n1;
+  }
+  return n2;
+}
 
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
-
-function findLongestWord() {}
-
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot",
+];
 
+function findLongestWord(words) {
+  let word;
+  if (words.length > 0) {
+    for (const separWord of words) {
+      word = !word || word.length < separWord.length ? separWord : word;
+    }
+  } else {
+    word = null;
+  }
+  return word;
+}
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
-
+function sumNumbers(numbers) {
+  return sum(numbers);
+}
 
 // Iteration #3.1 Bonus:
-function sum() {}
-
-
+function sum(numbers) {
+  let sum = 0;
+  for (const number of numbers) {
+    switch (typeof number) {
+      case "string":
+        // sum += !Number.isNaN(Number(number)) ? Number(number) : number.length;
+        sum += number.length;
+        break;
+      case "object":
+      case "array":
+        throw new Error("Unsupported data type sir or ma'am");
+      default:
+        sum += Number(number);
+        break;
+    }
+  }
+  return sum;
+}
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
-
+function averageNumbers(numbers) {
+  let sum = 0;
+  for (const number of numbers) {
+    sum += number;
+  }
+  const result = numbers.length > 0 ? sum / numbers.length : null;
+  return result;
+}
 
 // Level 2: Array of strings
-const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+const wordsArr = [
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace",
+];
 
-function averageWordLength() { }
+function averageWordLength(wordsArr) {
+  let sum = 0;
+  for (const word of wordsArr) {
+    sum += word.length;
+  }
+  const result = wordsArr.length > 0 ? sum / wordsArr.length : null;
+  return result;
+}
 
 // Bonus - Iteration #4.1
-function avg() {}
+function avg(mixArrs) {
+  let sum = 0;
+  for (const mixArr of mixArrs) {
+    if (typeof mixArr === "string") {
+      // sum += !Number.isNaN(Number(mixArr)) ? Number(mixArr) : mixArr.length;
+      sum += mixArr.length;
+    } else {
+      sum += Number(mixArr);
+    }
+  }
+  const result = mixArrs.length > 0 ? sum / mixArrs.length : null;
```

Comment:
```suggestion
  const result = mixArrs.length > 0 ? sum(mixArrs) / mixArrs.length : null;
```

And again and again... :D